source/core/src/main/com/csse3200/game/components/minigame/snake/Snake.java

```diff
@@ -0,0 +1,173 @@
+package com.csse3200.game.components.minigame.snake;
+
+import com.csse3200.game.components.minigame.Direction;
+import com.csse3200.game.components.minigame.Grid;
+import java.util.Deque;
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.ArrayList;
+
+/*
+ * Class for the snake in the snake game
+ */
+public class Snake {
+    private int x;
+    private int y;
+    private final Grid grid;
+    private final Deque<Segment> snakeBody;
+    private int length = 1;
+    private Direction direction;
+    private Direction nextDirection;
+    private final float movePeriod;
+    private float moveTimer;
+
+    public Snake(Grid grid, int startX, int startY, Direction startDirection, int startLength, float movePeriod) {
+        this.grid = grid;
+        this.movePeriod = movePeriod;
+        this.moveTimer = movePeriod;
+        this.direction = startDirection;
+        this.nextDirection = startDirection;
+        this.length = startLength;
+        x = startX;
+        y = startY;
+        snakeBody = new ArrayDeque<>();
+    }
+
+    /*
+     * Sets the direction the snake is moving
+     * @param direction: the direction to set to 
+     */
+    public void setDirection(Direction direction) {
+        this.direction = direction;
+    }
+
+    /*
+     * Get the current direction of the snake
+     */
+    public Direction getDirection() {return this.direction;}
+
+    /*
+     * Update snake direction with checks
+     * @param direction: the direction to update to
+     */
+    public void updateDirectionOnInput(Direction direction) {
+        if (direction == Direction.UP && this.direction != Direction.DOWN) {
+            this.nextDirection = direction;
+        }
+        if (direction == Direction.DOWN && this.direction != Direction.UP) {
+            this.nextDirection = direction;
+        }
+        if (direction == Direction.LEFT && this.direction != Direction.RIGHT) {
+            this.nextDirection = direction;
+        }
+        if (direction == Direction.RIGHT && this.direction != Direction.LEFT) {
+            this.nextDirection = direction;
+        }
+    }
+
+    /*
+     * moves the snake in a direction
+     */
+    public void move(Direction direction) {
+        snakeBody.add(new Segment(x, y));
```

Comment:
method needs unit tests

Reply:
It already has one, will be in next pull request